lib/LaTeXML/Core/Tokens.pm

```diff
@@ -37,16 +37,23 @@ sub Tokens {
 # .... Efficiently! since this seems to be called MANY times.
 sub new {
   my ($class, @tokens) = @_;
-  my $r;
-  return bless [map { (($r = ref $_) eq 'LaTeXML::Core::Token' ? $_
-        : ($r eq 'LaTeXML::Core::Tokens' ? @$_
-          : Fatal('misdefined', $r, undef, "Expected a Token, got " . Stringify($_)))) }
-      @tokens], $class; }
+  my ($r, $marks);
+  if (ref $tokens[-1] eq 'ARRAY') {
+    $marks = pop @tokens; }
+  @tokens = map { (($r = ref $_) eq 'LaTeXML::Core::Token' ? $_
+      : ($r eq 'LaTeXML::Core::Tokens' ? @$_
+        : Fatal('misdefined', $r, undef, "Expected a Token, got " . Stringify($_)))) }
+    @tokens;
+  push(@tokens, $marks) if $marks;
+  return bless [@tokens], $class; }
 
 # Return a list of the tokens making up this Tokens
 sub unlist {
   my ($self) = @_;
-  return @$self; }
+  my @unlisted = @$self;
+  if (ref $unlisted[-1] eq 'ARRAY') {
+    @unlisted = @unlisted[0 .. scalar(@unlisted) - 2]; }
```

Comment:
So, `->unlist` *removes* the no-expand-once mask?

Reply:
Ignores rather. I use a slice to copy the array without the last element. The tokens object still retains the masks and they can be accessed after the unlist.

Reply:
Oooooh, so you may have *both* array-wrapped tokens *and* a mask (both of which inhibit expansion?

Reply:
The array-wrapped tokens would cause a Fatal error when `Tokens()` is called, they're not allowed in as legal members. Those need to be unwrapped immediately after they are returned by `readToken` or `readXToken`, which are the only two producers of them.

Reply:
OK, but my original point was that any caller of `->unlist` gets the no-expand-once flags/mask/marks removed, right?
(which implies that callers are expected not to expect it, and any caller that expects those masks preserved has to make its own arrangements;  that is to say that `unlist` is no longer "merely" an accessor)

Reply:
I am unsure why you are searching for new behaviors of unlist? The code change here is exactly to keep it _the same_ it has always been, guarding against the new special case where there is something different at the last slot of the array. It continues to return the array of tokens, as it always has.

If someone needs the masks explicitly, there is a new accessor:
```perl
my @marks = $tokens->get_marks;
```

Though I should really decide if it should be marks or masks... In general I am abusing the array for behavior that is really meant to be in a separate slot. Refactoring to a newly added `MaskedTokens` could improve on that front.

Reply:
I'm testing an alternative approach to how no-expand-once (or whatever it should be called) due to `\the` is noted. Thus I'm trying to track down the places where such a flag/mark/mask are preserved and where they are dropped.

I'm inclined to think that a MaskedToken (or rather a new CC) is a better approach than two distinct mechanisms (whether smuggled or renamed to MaskedTokens), but first I have to understand and replicate the current behaviour.  Hence the questions.

Reply:
Cool! The main place of import is:
 - `readToken`, `readXToken` and `readBalanced` in the `DefExpanded` and `DefPlain` parameter types for the defining primitives need to preserve and pass on these marks. The only real place they are used is in the main logic of creating a new Expandable through `do_def`, where this masks-based neutralization is crucial.

For all non-definitional uses everything is handled internally by `readXToken`, since a loop that keeps calling it will stop short of expanding an array-wrapped token, and will return it as-is, unwrapped.

I think if those two aspects work, an alternative approach should work too.